### PR TITLE
chore: extract error-boundary

### PR DESCRIPTION
**Pull request by gabriel-mirantis**

### CODE REVIEW

File: packages/ui-components/error-boundary/.swcrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      "dynamicImport": false
+    },
+    "transform": {
+      "legacyDecorator": true,
+      "decoratorMetadata": true
+    },
+    "target": "es2019"
+  }
+}
+
```
Comment (jansav) [inline packages/ui-components/error-boundary/.swcrc:19]: I don't think we are using .swc here?

File: packages/ui-components/error-boundary/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# @k8slens/error-boundary
+
+This package contains stuff related to creating Lens-applications.
```
Comment (jansav) [inline packages/ui-components/error-boundary/README.md:3]: Copypaste? ;)

File: packages/ui-components/error-boundary/.swcrc (hunk 1 above)
Comment (gabriel-mirantis) [reply packages/ui-components/error-boundary/.swcrc:19]: there was an error until I added this file.  I can go re-create the error

File: packages/ui-components/error-boundary/README.md (hunk 2 above)
Comment (gabriel-mirantis) [reply packages/ui-components/error-boundary/README.md:3]: naturally

Comment (gabriel-mirantis) [reply packages/ui-components/error-boundary/README.md:3]: noop.  does not look like it is needed.  it is removed now

**Review by jansav — APPROVED** [review]

**Pull request merged**